lib/puppet/network/http/rack/rest.rb

```diff
@@ -103,7 +103,21 @@ def client_cert(request)
     # NOTE: The SSL_CLIENT_CERT environment variable will be the empty string
     # when Puppet agent nodes have not yet obtained a signed certificate.
     if cert.nil? || cert.empty?
-      nil
+      # When running with unicorn, the SSL_CLIENT_CERT variable is not available
+      # in the environment, therefore we have to pass a header: 'X-SSL-Client-Cert'
+      cert = request.env['HTTP_X_SSL_CLIENT_CERT']
```

Comment:
If this value is sent as a header, is it possible to explicitly pull this value with `request.params['HTTP_X_SSL_CLIENT_CERT']` instead of using the environment? This will make the code more clear on its intent.